Extension/src/LanguageServer/protocolFilter.ts

```diff
@@ -49,7 +49,12 @@ export function createProtocolFilter(me: Client, clients: ClientCollection): Mid
 
         provideCompletionItem: invoke4,
         resolveCompletionItem: invoke2,
-        provideHover: invoke3,
+        provideHover: (document, position, token, next: (document, position, token) => any) => {
+            if (clients.checkOwnership(me, document)) {
```

Comment:
Are there any other lsp messages that need to be filtered in this way?

Reply:
Maybe later -- the formatting scenario is pretty busted with multi-root, but there are a bunch of issues on that for later (not worth dealing with now).